src/widgets/NumericTextCtrl.cpp

```diff
@@ -592,33 +601,38 @@ void NumericTextCtrl::OnContext(wxContextMenuEvent &event)
    // user happens to check the first menuitem and then is 
    // moving down the menu when the ...CTRL_UPDATED event
    // handler kicks in.
-   for (i = 0; i < GetNumBuiltins(); i++) {
-      if (menu.IsChecked(ID_MENU + i) && i != currentSelection) {
-         SetFormatString(GetBuiltinFormat(i));
-      
-         int eventType = 0;
-         switch (mType) {
-            case NumericConverterType::TIME:
-               eventType = EVT_TIMETEXTCTRL_UPDATED;
-               break;
-            case NumericConverterType::FREQUENCY:
-               eventType = EVT_FREQUENCYTEXTCTRL_UPDATED;
-               break;
-            case NumericConverterType::BANDWIDTH:
-               eventType = EVT_BANDWIDTHTEXTCTRL_UPDATED;
-               break;
-            default:
-               wxASSERT(false);
-               break;
-         }
-      
-         wxCommandEvent e(eventType, GetId());
-         e.SetInt(i);
```

Comment:
The integer information is no longer set.  So I made a careful check of all old uses of these events to be sure the integer value is never retrieved from the event.

But there IS one such place still!  In `SpectralSelectionBar::OnUpdate`

But then it's only this inialization `int index = evt.GetInt();` of a variable that is never used, so delete that please

Reply:
If there were an Observer message with only the necessary information, I would not have needed to check all that.  But you would have written much more stuff.